src/Microsoft.DotNet.Wpf/src/PresentationFramework/System/Windows/Documents/SelectionWordBreaker.cs

```diff
@@ -260,9 +260,9 @@ private static CharClass[] GetClasses(char[] text)
         // Returns true if a char is a non-spacing diacritic or kashida.
         private static bool IsDiacriticOrKashida(char ch)
         {
-            UInt16 []charType3 = new UInt16[1];
```

Comment:
In all the places that calls `GetStringTypeEx`, they all seem to pass either 1 or 2 items. I think it would be cleaner if you wrapped local variables in a span instead of stackallocating the span.

Something like this here:
```
UInt16 charType3 = 0;

SafeNativeMethods.GetStringTypeEx(0 /* ignored */, SafeNativeMethods.CT_CTYPE3, new ReadOnlySpan<char>(in ch), new Span<UInt16>(ref charType3));
```

You would then replace usages of charType3[0] to use charType3 directly.

Reply:
Good point, we can certainly do that here, addressed.